rotkehlchen/tasks/calendar.py

```diff
@@ -75,123 +78,206 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
         )
 
 
-def maybe_create_ens_reminders(database: DBHandler) -> None:
-    """Check ENS registration and renewal history events and create reminders if needed."""
-    db_history_events = DBHistoryEvents(database=database)
-    ens_events: list[EvmEvent] = []
-    with database.conn.read_ctx() as cursor:
-        for event_type, event_subtype in (
-            (HistoryEventType.TRADE, HistoryEventSubType.SPEND),
-            (HistoryEventType.RENEW, HistoryEventSubType.NONE),
-        ):
-            ens_events.extend(db_history_events.get_history_events(
-                cursor=cursor,
-                has_premium=True,  # not limiting here
-                group_by_event_ids=False,
-                filter_query=EvmEventFilterQuery.make(
-                    and_op=True,
-                    counterparties=[CPT_ENS],
-                    event_types=[event_type],
-                    event_subtypes=[event_subtype],
-                ),
-            ))
+class CalendarReminderCreator(CustomizableDateMixin):
```

Comment:
Add a docstring for this class. Specify that it's supposed to be a short-lived object to help with creating calendar reminders.